packages/xchain-thorchain/src/client.ts

```diff
@@ -382,20 +389,116 @@ class Client extends BaseXChainClient implements ThorchainClient, XChainClient {
-    const txData: TxData | null = txResult && txResult.logs ? getDepositTxDataFromLogs(txResult.logs, address) : null
-    if (!txResult || !txData) throw new Error(`Failed to get transaction data (tx-hash: ${txId})`)
-
-    const { from, to, type } = txData
-
-    return {
+    try {
+      const txResult = await this.cosmosClient.txsHashGet(txId)
+      const bond = txResult.logs && txResult.logs[0].events.filter((i) => i.type === 'bond')
+      const transfer = txResult.logs && txResult.logs[0].events.filter((i) => i.type === 'transfer')
+      if (!transfer) throw new Error(`Failed to get transaction logs (tx-hash: ${txId})`)
+
+      const sender = transfer[0].attributes.find((attr) => attr.key === 'sender')?.value
+      const senderAmount = transfer[0].attributes.filter((attr) => attr.key === 'amount')[1].value
+
+      const assetString = senderAmount.split(/(?<=\d)(?=[a-zA-Z])/)
+      const senderAsset = assetString[1] === 'rune' ? AssetRuneNative : assetFromStringEx(assetString[1])
+
+      const senderAddress = sender ? sender : address
+      const message = txResult.logs && txResult.logs[0].events.filter((i) => i.type === 'message')
+      const coinSpent = txResult.logs && txResult.logs[0].events.filter((i) => i.type === 'coin_spent')
+      if (!message || !coinSpent) throw new Error(`Failed to get transaction logs (tx-hash: ${txId})`)
+
+      const action = message[0].attributes.find((attr) => attr.key === 'action')?.value
+      if (!bond) throw new Error(`Failed to get transaction logs (tx-hash: ${txId})`)
+      // Rune only transactions
+      if (bond[0].type === 'bond' || action === 'send') {
+        const assetTo = AssetRuneNative
+        const txData: TxData | null =
+          txResult && txResult.logs
+            ? getDepositTxDataFromLogs(txResult.logs, senderAddress, senderAsset, assetTo)
+            : null
+        //console.log(JSON.stringify(txData, null, 2))
+        if (!txData) throw new Error(`Failed to get transaction data (tx-hash: ${txId})`)
+
+        const { from, to, type } = txData
+        return {
+          hash: txId,
+          asset: senderAsset,
+          from,
+          to,
+          date: new Date(txResult.timestamp),
+          type,
+        }
+      }
+      // synths and other tx types
+      const messageBody = JSON.stringify(txResult.tx?.body.messages).split(':')
+      const assetTo = assetFromStringEx(messageBody[7])
+
+      const txData: TxData | null =
+        txResult && txResult.logs ? getDepositTxDataFromLogs(txResult.logs, senderAddress, senderAsset, assetTo) : null
+      if (!txData) throw new Error(`Failed to get transaction data (tx-hash: ${txId})`)
+
+      const { from, to, type } = txData
+
+      return {
+        hash: txId,
+        asset: senderAsset,
+        from,
+        to,
+        date: new Date(txResult.timestamp),
+        type,
+      }
+    } catch {
+      return await this.getTransactionDataThornode(txId)
```

Comment:
if this call fails, there is nothing to catch it. should this be here?

Reply:
yeah true, I move it out of there